PlayerAPI/Card.cpp

```diff
@@ -1,96 +1,82 @@
-/*
- * Card Class
- *
- * -David Browning, Ligia Frangello & Katie Sweet
- *
- * The purpose of this file is to allow others to build a card object.
- * Cards contain a suit and a value.
- * Suit is an enum {HEARTS}
- *
- * The constuctor accepts the suit enum and one integer.
- *
- * The suit enum (HEARTS, SPADES, CLUBS, DIAMONDS, UNDEFINED) determines the
- * suit. The reason for the UNDEFINED suit type is that in the Player Class,
- * there is a function called requestMove() that returns a vector of Cards.
- * In the game Crazy Eight's, a move can consist of drawing a card from the
- * deck. We need an UNDEFINED type to alertnthe Crazy Eight's game logic that
- * they chose the draw pile instead of playing a card.
- *
- * The integer (1-13) determines the value. Value is on a 1 based index for
- * easier scoring.
- *
- */
-
+#include "Card.hpp"
+#include <algorithm>
 #include <iostream>
 #include <iterator>
 #include <random>
-#include <vector>
 
-enum Suit
+Card::Card(Suit su)
 {
-  HEARTS,
-  SPADES,
-  CLUBS,
-  DIAMONDS,
-  UNDEFINED
-};
+  suit = su;
+  if (su == UNDEFINED)
+  {
+    value = 0;
+  }
+  else
+  {
+    std::cout << "Error. Cannot instantiate a card without a value."
+              << std::endl;
+    throw(2);
+  }
+}
 
-class Card
+Card::Card(Suit su, int val)
```

Comment:
Please change this to an initializer list rather than assigning in the body of the constructor